catalog/resource_sql_table.go

```diff
@@ -109,6 +120,11 @@ func (a SqlTablesAPI) getTable(name string) (ti SqlTableInfo, err error) {
 	return
 }
 
+func (a SqlTablesAPI) getPartitions(name string) (ti PartitionInfo, err error) {
```

Comment:
Can we use the Go SDK here with `w.Tables.Get()`?

Reply:
we cannot, because the PartitionIndex is specified as `json:"partition_index,omitempty"`, so we cannot distinguish between 0 & null

Reply:
The Go SDK will include fields set explicitly to zero in `ForceSendFields` here: https://github.com/databricks/databricks-sdk-go/blob/515359fde79fcfba2dbe3024a4ed7dfd4fa42818/service/catalog/model.go#L499. If `0` is explicitly sent back from the API, `ForceSendFields` will include `PartitionIndex`.

Reply:
rewritten to use go sdk 👍